pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -935,14 +939,9 @@ protected boolean commitSegment(String controllerVipUrl, boolean isSplitCommit)
     return segmentCommitter.commit(_segmentBuildDescriptor);
   }
 
-  protected boolean buildSegmentAndReplace() {
+  protected void buildSegmentAndReplace() throws Exception {
     SegmentBuildDescriptor descriptor = buildSegmentInternal(false);
```

Comment:
(major) when segment build failed, it will return `null` descriptor instead of throwing exception. In order to make it work, we need to change `buildSegmentInternal()` to throw exception as well

Reply:
yes. `buildSegmentInternal` has already been changed to throw exception.

Reply:
Sorry I missed that. When we throw the exception, we should not log the error. The catcher is responsible for logging the error, or we will log multiple errors for the same exception

Reply:
yup. working on it.

Reply:
done